internal/command/deploy/deploy.go

```diff
@@ -250,48 +249,20 @@ func useMachines(ctx context.Context, appConfig *appconfig.Config, appCompact *a
 // determineAppConfig fetches the app config from a local file, or in its absence, from the API
 func determineAppConfig(ctx context.Context) (cfg *appconfig.Config, err error) {
 	tb := render.NewTextBlock(ctx, "Verifying app config")
-	client := client.FromContext(ctx).API()
 	appNameFromContext := appconfig.NameFromContext(ctx)
 	if cfg = appconfig.ConfigFromContext(ctx); cfg == nil {
 		logger := logger.FromContext(ctx)
 		logger.Debug("no local app config detected; fetching from backend ...")
 
-		var apiConfig *api.AppConfig
-		if apiConfig, err = client.GetConfig(ctx, appNameFromContext); err != nil {
-			err = fmt.Errorf("failed fetching existing app config: %w", err)
-			return
-		}
-
-		basicApp, err := client.GetAppBasic(ctx, appNameFromContext)
-		if err != nil {
-			return nil, err
-		}
-
-		cfg, err = appconfig.FromDefinition(&apiConfig.Definition)
+		cfg, err = appconfig.FromRemoteApp(ctx, appNameFromContext)
 		if err != nil {
-			return nil, fmt.Errorf("Failed to convert definition into config: %w", err)
+			return
 		}
 
-		cfg.AppName = basicApp.Name
-
-		if err := cfg.SetPlatformVersion(basicApp.PlatformVersion); err != nil {
-			return cfg, err
-		}
 	} else {
-		parsedCfg, err := client.ParseConfig(ctx, appNameFromContext, cfg.SanitizedDefinition())
+		err, _ = cfg.Validate(ctx)
```

Comment:
This is looking great!  my last take on it would be that cfg validation should happen as last step before return from this function.  A line above where we print the config was verified to be valid.

Reply:
Do you mean moving the cfg.Validate line to literally right before?

Reply:
I mean, before this line https://github.com/superfly/flyctl/blob/e363285490718b10adf35c87e976d503f7f3af69/internal/command/deploy/deploy.go#L289

Reply:
the point is that any change we do to the config, like updating envvars, region and whatever comes next, has to be considered for validation

Reply:
oh i see what you mean! yea i understand